contrib/python/src/python/pants/contrib/python/checks/tasks/checkstyle/checkstyle.py

```diff
@@ -94,11 +90,16 @@ def checker_pex(self, interpreter):
     else:
       checker_id = hash_all([self._CHECKER_REQ])
 
+    # TODO(CMLivingston): We should be able to build a multi interpreter pex here and avoid
+    # multiple pexes and workunits.
+    # Address in: <ticket>
     pex_path = os.path.join(self.workdir, 'checker', checker_id, str(interpreter.identity))
+
     if not os.path.exists(pex_path):
       with self.context.new_workunit(name='build-checker'):
         with safe_concurrent_creation(pex_path) as chroot:
           builder = PEXBuilder(path=chroot, interpreter=interpreter)
+          builder.add_interpreter_constraint(str(interpreter.identity.requirement))
```

Comment:
This should not be needed. We pass an explicit interpreter to all build steps and use the same one for the run step (`PEX(... interpreter=interpreter)`). Can you dig and figure out what is going on here?

Reply:
This is needed to guard against the case where a user has a pexrc with PEX_PYTHON or PEX_PYTHON_PATH set. If no constraints are supplied, pex will exec with the lowest compatible interpreter, hence python2.7 executing for a python 3 pex file.

Reply:
Hrm - that's seems like a bug. If a PEX() is constructed with a specific interpreter it should probably use that interpreter and ignore pexrc, etc. This seems like an out of scope API hole, but lots of pants code is currently broken I think in the face of a user with a pexrc. Thanks for the explanation, I'll file an issue.

Reply:
https://github.com/pantsbuild/pex/issues/604